src/main/java/mate.academy/lib/Injector.java

```diff
@@ -1,13 +1,84 @@
 package mate.academy.lib;
 
+import java.lang.reflect.Constructor;
+import java.lang.reflect.Field;
+import java.util.HashMap;
+import java.util.Map;
+import mate.academy.service.FileReaderService;
+import mate.academy.service.ProductParser;
+import mate.academy.service.ProductService;
+import mate.academy.service.impl.FileReaderServiceImpl;
+import mate.academy.service.impl.ProductParserImpl;
+import mate.academy.service.impl.ProductServiceImpl;
+
+/**
+ * If the class is annotated with the @Component annotation, then we can create an implementation
+ * If not, throw an error
+ * Check if an instance has been created before creating a new instance
+ */
```

Comment:
FYI - bad javadoc 
See more info about javadoc and how to write it properly if you want to use it